src/librustc_typeck/collect.rs

```diff
@@ -668,7 +668,7 @@ fn adt_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 /// Ensures that the super-predicates of the trait with def-id
 /// trait_def_id are converted and stored. This also ensures that
 /// the transitive super-predicates are converted;
-fn super_predicates_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+pub fn super_predicates_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
Why is this public now? It's a provider so I think it should not have to be, we should be using the query instead.

Reply:
It's public because I needed it in the hack that I added to astconv, which you correctly pointed out just above is the wrong place 😛